README.md

```diff
@@ -10,15 +10,42 @@ Go clients for talking to a [kubernetes](http://kubernetes.io/) cluster.
 
-client-go has the same release cycle as the Kubernetes main repository. For example, in the 1.4 release cycle, the contents in `1.4/` folder are subjected to changes. Once 1.4 is released, new changes will go into the `1.5/` folder. We will make great efforts to not change the public interface of a version of the client once that version has been released. We may change the interface between versions. Old versions of the client will be retained for two release cycles.
+The master branch will track the head in the main Kubernetes repo and accumulating changes. We will bump the client-go version when:
+* A new major (very rare) or a new minor version is released in the Kubernetes main repository.
+* A feature in the master branch is popular and users want it in a stable branch. 
+
+Depending on the nature of the changes accumulated when we bump the version, we will bump version number according to semver.
+
+Note that these are the versions *of the client library itself*, not the versions of the Kubernetes clusters which the library will talk to. 
+
+#### Compatibility: Your code <-> client-go
+
+`client-go` will follow [semver](http://semver.org/), so your code keeps compatible until the major version of client-go gets bumped.
+
+#### Compatibility: client-go <-> Kubernetes clusters
+
+`client-go` versions will be backwards compatible with many Kubernetes clusters. As we increment `client-go` versions, we will note which Kubernetes versions we expect them to work with. 
+
+We do not currently plan to back-port new Kubernetes features into older clients, but we can consider this if some version of the client becomes especially popular. We'll revisit this when we have accumulated a few versions. You can check out the [CHANGELOG](./CHANGELOG.md) for notable changes.
+
+#### Compatibility matrix
+
+* **client-go/1.4** is compatible with Kubernetes 1.3 through 1.5, and possibly further; it includes all features provided by Kubernetes 1.4.
+* **client-go/1.5** is compatible with Kubernetes 1.3 through 1.5, and possibly further; it includes all features provided by Kubernetes 1.4
+
+#### Why do the 1.4 and 1.5 branch contain top-level folder named after the version?
+
+`client-go` used to include different versions in the master branch, stored in top-level folders. 1.4 and 1.5 branch keep the top-level folders so they do not break the import lines of existing users. These top-level folders are removed from the master branch and future branches.
 
 ### How to get it
 
-You can `go get` to get a release of client-go, e.g., `go get k8s.io/client-go/1.4/...` or `go get k8s.io/client-go/1.4/kubernetes`.
+If you use `go get` to get client-go, **you will get the unstable master branch!** You should `git checkout` a stable branch.
```

Comment:
Instead of "You should `git checkout` a stable branch." I would say "Instead, you can checkout a stable branch using `git checkout` or vendor tool ..."